tensorflow/python/kernel_tests/random/random_gamma_test.py

```diff
@@ -168,6 +170,7 @@ def testCPUGPUMatch(self):
       else:
         self.assertAllClose(results[False], results[True], rtol=1e-6, atol=1e-6)
 
+  @test_util.run_deprecated_v1
```

Comment:
If we wrap the test code inside a tf.function, does it work correctly? That would be preferrable to forcing V1 mode.

Something like this:

```
@tf.function
def test_fn():
    x_np = array_ops.ones([batch_size, 4, 6, 3])
    x_out = array_ops.ones([batch_size, 2, 3, 12])
    self._testOne(x_np, block_size, x_out)
    return 1
```

Reply:
Change in space to depth is just utilizing the common code which is already present and used in other tests.

Reply:
Also pushed a further change in space to depth so that NHWC inputs testing is just not restricted to CPU but to any device.